File: 07-interfaces/python/aguila.py
"""
aguila.py - Ave que vuela pero NO nada

Junto con el pinguino demuestra que las dos capacidades son
independientes entre si, e independientes de la jerarquia.
"""

from ave import Ave
from volador import Volador


class Aguila(Ave, Volador):

    def __init__(self, nombre, edad):
        super().__init__(nombre, edad, 7000)

    def sonido(self):
        return "Screech"

    # --- contrato Volador ---
    def volar(self):
        print(f"   {self._nombre} planea sobre la cordillera")

    def altura_maxima(self):
        return 6000

File: 07-interfaces/java/Aguila.java
/*
 * Aguila.java - Ave que vuela pero NO nada
 *
 * Junto con el pinguino demuestra que las dos capacidades son
 * independientes entre si, e independientes de la jerarquia.
 */
public class Aguila extends Ave implements Volador {

    public Aguila(String nombre, int edad) {
        super(nombre, edad, 7000);
    }

    @Override public String sonido() { return "Screech"; }

    // --- contrato Volador ---
    @Override public void volar() {
        System.out.println("   " + nombre + " planea sobre la cordillera");
    }
    @Override public int alturaMaxima() { return 6000; }
}
